index.js

```diff
@@ -34,10 +35,12 @@ function externalLinks(options) {
       var protocol = ctx.url.slice(0, ctx.url.indexOf(':'))
       var data
       var props
+      var children
 
       if (isAbsoluteURL(ctx.url) && protocols.indexOf(protocol) !== -1) {
         data = node.data || (node.data = {})
         props = data.hProperties || (data.hProperties = {})
+        children = data.hChildren || (data.hChildren = node.children)
```

Comment:
Unfortunately `data.hChildren = node.children` will not work in most cases. It does in this test case though, because there’s only a text node, and text nodes overlap between hast and mdast. But if there would be richer content, such as emphasis or inline code, it would break 😬

We can do two things:

1. ask people to provide mdast content instead of hast content
2. add a new mdast node that’s typically not rendered (empty text? unknown parent node?) with the hast content as `hChildren`